src/compiler/scala/tools/nsc/Reporting.scala

```diff
@@ -113,14 +113,19 @@ trait Reporting extends scala.reflect.internal.Reporting { self: ast.Positions w
       reportedFeature += featureTrait
 
       val msg = s"$featureDesc $req be enabled\nby making the implicit value $fqname visible.$explain" replace ("#", construct)
+      // maybe pos.source.file.file.getParentFile.getName or Path(source.file.file).parent.name
```

Comment:
This should be a path test, like the current test of the path string, but `Path.apply` does `File.isFile`, and also the `parent` method doesn't inspire confidence. Using the underlying `java.io.File` means that `VirtualFile` is treated differently and fails the test. Luckily this doesn't matter for the current application. But it would be natural for sbt to supply a `VirtualFile`.

Reply:
Just use Java's `Path`, for which `getParent` can be null.